torch/csrc/autograd/functions/special.cpp

```diff
@@ -1,10 +1,12 @@
+#ifndef NO_PYTHON
+#include <Python.h>
+#endif
```

Comment:
It doesn't look like we're using Python in this file anymore. Is this needed?

Reply:
You're right, although I had to add this one back to prevent an XOPEN_SOURCE warning. I'll take a closer look.

Reply:
Ok, no warnings anymore.